### PR TITLE
Replace subsampleCounts with rarefyAssay

### PR DESCRIPTION
**Pull request by theotimeee**
After mia PR https://github.com/microbiome/mia/pull/536 that renamed the following function : 

- `subsampleCounts` --> `rarefyAssay`

This PR replaces `subsampleCounts` with `rarefyAssay` in the OMA book.

### CODE REVIEW

**Comment by antagomir** [comment]
Seems good.

File: inst/pages/12_quality_control.qmd [hunk 1]
```diff
@@ -301,7 +301,7 @@ plotColData(tse,"sum","SampleType", colour_by = "SampleType") +
 ```
 
 In addition, data can be rarefied with
-[subsampleCounts](https://microbiome.github.io/mia/reference/subsampleCounts.html),
+[rarefyAssay](https://microbiome.github.io/mia/reference/rarefyAssay.html),
```
Comment (antagomir) [inline inst/pages/12_quality_control.qmd:328]: I would also add that "This remains controversial, however, and strategies to mitigate the information loss in rarefaction have been proposed [@citations]."

Add these papers in the bibtext file and cite:
https://journals.asm.org/doi/10.1128/msphere.00354-23
https://journals.asm.org/doi/10.1128/msphere.00355-23

**Review by antagomir — CHANGES_REQUESTED** [review]

File: inst/assets/bibliography.bib [hunk 2]
```diff
@@ -2139,3 +2139,25 @@ @article{Geistlinger2023
     url = {https://doi.org/10.1038/s41587-023-01872-y},
     year = {2023}
 }
+
+@article{Schloss2024rarefaction1,
+    author = {Patrick D. Schloss},
+    title = {Rarefaction is currently the best approach to control for uneven sequencing     effort in amplicon sequence analyses},
+    journal = {mSphere},
```
Comment (antagomir) [inline inst/assets/bibliography.bib]: Put this in double curly brackets to ensure that the capitalization will be maintained in rendering to other formats. Like {{mSphere}} (this is a LaTeX trick)

File: inst/assets/bibliography.bib [hunk 3]
```diff
@@ -2139,3 +2139,25 @@ @article{Geistlinger2023
     url = {https://doi.org/10.1038/s41587-023-01872-y},
     year = {2023}
 }
+
+@article{Schloss2024rarefaction1,
+    author = {Patrick D. Schloss},
+    title = {Rarefaction is currently the best approach to control for uneven sequencing     effort in amplicon sequence analyses},
```
Comment (antagomir) [inline inst/assets/bibliography.bib]: remove extra spaces

File: inst/assets/bibliography.bib [hunk 4]
```diff
@@ -2139,3 +2139,25 @@ @article{Geistlinger2023
...
+    journal = {mSphere},
+    volume = {9},
+    number = {2},
+    pages = {e00354-23},
+    url = {https://journals.asm.org/doi/abs/10.1128/msphere.00354-23}
```
Comment (antagomir) [inline inst/assets/bibliography.bib]: Change this toi doi instead.

doi = {10.1128/msphere.00354-23},

File: inst/assets/bibliography.bib [hunk 5]
```diff
@@ -2139,3 +2139,25 @@ @article{Geistlinger2023
...
+}
+
+@article{Schloss2024rarefaction2,
+    author = {Patrick D. Schloss},
+    title = {Waste not, want not: revisiting the analysis that called into question the practice of rarefaction},
+    journal = {mSphere},
+    volume = {9},
+    number = {1},
+    pages = {e00355-23},
+    url = {https://journals.asm.org/doi/abs/10.1128/msphere.00355-23}
```
Comment (antagomir) [inline inst/assets/bibliography.bib]: change into doi

File: inst/assets/bibliography.bib [hunk 6]
```diff
@@ -2139,3 +2139,25 @@ @article{Geistlinger2023
...
+    number = {2},
+    pages = {e00354-23},
+    url = {https://journals.asm.org/doi/abs/10.1128/msphere.00354-23}
+    year = {2024}
+}
+
+@article{Schloss2024rarefaction2,
+    author = {Patrick D. Schloss},
+    title = {Waste not, want not: revisiting the analysis that called into question the practice of rarefaction},
+    journal = {mSphere},
```
Comment (antagomir) [inline inst/assets/bibliography.bib]: {{}}

**Review by 0xMuluh — COMMENTED** [review]
> LGTM

**Review by TuomasBorman — APPROVED** [review]

**Pull request merged**